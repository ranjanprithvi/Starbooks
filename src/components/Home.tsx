import {
    Heading,
    VStack,
    Image,
    GridItem,
    Box,
    HStack,
} from "@chakra-ui/react";
import logo from "../assets/Logo.png";

const Home = () => {
    return (
        <GridItem colSpan={2} height="100%">
            <VStack
                height="100%"
                width="53vw"
                justifyContent="center"
                marginX="auto"
            >
                <Image src={logo} height="48" />
                <Heading>Welcome to Starbooks!</Heading>
                <VStack>
                    <Box>
                        Starbooks is a library management app where visitors can
                        view the books catalog, members can view their details
                        and active rentals after logging in, and admins can
                        manage the books catalog, users, and rentals.
                    </Box>
                    <br></br>
                </VStack>
            </VStack>
        </GridItem>
    );
};

export default Home;
