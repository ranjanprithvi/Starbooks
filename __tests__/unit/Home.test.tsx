import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Home from "../../src/components/Home";

test("Renders the home page", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
});
