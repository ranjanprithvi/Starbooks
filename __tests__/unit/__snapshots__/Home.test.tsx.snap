// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the home page 1`] = `
<div
  className="css-19x3waq"
>
  <div
    className="chakra-stack css-18x02rm"
  >
    <img
      className="chakra-image css-1jsohtw"
      src="test-file-stub"
    />
    <h2
      className="chakra-heading css-0"
    >
      Welcome to Starbooks!
    </h2>
    <div
      className="chakra-stack css-tl3ftk"
    >
      <div
        className="css-0"
      >
        Starbooks is a library management app where visitors can view the books catalog, members can view their details and active rentals after logging in, and admins can manage the books catalog, users, and rentals.
      </div>
      <br />
    </div>
  </div>
</div>
`;
